### PR TITLE
Feature : Wishlist Management

### PR DESCRIPTION
**Pull request by yogesh-**
Wishlist Feature 
- Add to wishlist
- remove from wishlist
- Add to wishlist from cart

Preview Link - https://deploy-preview-6--sneakerzz.netlify.app/

https://user-images.githubusercontent.com/7193961/161573227-c56c11f5-cbaf-4d72-bc1b-c5b5592955d2.mp4

### CODE REVIEW

File: src/context/wishContext.jsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React, { useContext, createContext, useReducer } from "react";
+import { wishReducer } from "../reducers/wishReducer";
+
+const wishListContext = createContext();
+
+const WishListProvider = ({ children }) => {
+  const [wishState, wishDispatch] = useReducer(wishReducer, {
+    wishCount: 0,
```
Comment (chanchal-panpaliya) [inline src/context/wishContext.jsx:8]: alternative way - in return section just add - <span> {wishlist.length} </span> . so , don't need to add wishCount:0.

Comment (yogesh-) [reply src/context/wishContext.jsx:8]: wow, good suggestion.

**Review by chanchal-panpaliya — COMMENTED** [review]
code is good . format your code - remove extra space , console .

File: src/pages/products/products.jsx [hunk 2]
```diff
@@ -45,6 +47,8 @@ export const Products = () => {
   const getFinalItems = sortItems(getRatedItems, state.sortBy);
 
   console.log(load);
+  console.log(wishDispatch, "from wishdispatch");
+  console.log(cartDispatch, "from cartDispatch");
```
Comment (tushargoyal812) [inline src/pages/products/products.jsx:51]: remove console

File: src/components/navbar/navbar.jsx [hunk 3]
```diff
@@ -2,9 +2,11 @@ import React from "react";
 import { Link } from "react-router-dom";
 import "./navbar.css";
 import { useCart } from "../../context/cartContext";
+import { useWish } from "../../context/wishContext";
```
Comment (sharmayatendra) [inline src/components/navbar/navbar.jsx:5]: you can put multiple context file into single index file & import it from there

Comment (yogesh-) [reply src/components/navbar/navbar.jsx:5]: ok

File: src/pages/products/products.jsx (hunk 2 above)
Comment (yogesh-) [reply src/pages/products/products.jsx:51]: done

**Review by sharmayatendra — APPROVED** [review]
on the product listing page after clicking add to wishlist button it should ideally change to remove from wishlist or hidden coz it might create confusion for user.
when i clicked on add to cart from wishlist the cart items count is showing including quantity you can look into that
good work though

**Pull request merged**